### PR TITLE
feat: add get_permission_info to resource reader mixin + readers

### PR DESCRIPTION
**Pull request by EmanuelCampos**

### CODE REVIEW

File: llama-index-core/llama_index/core/readers/base.py [hunk 1]
```diff
@@ -82,6 +82,20 @@ async def alist_resources(self, *args: Any, **kwargs: Any) -> List[str]:
         """
         return self.list_resources(*args, **kwargs)
 
+    @abstractmethod
+    def get_permission_info(self, resource_id: str, *args: Any, **kwargs: Any) -> Dict:
```
Comment (logan-markewich) [inline llama-index-core/llama_index/core/readers/base.py:86]: This breaks quite a few classes it seems, looking at the tests

**Comment by logan-markewich** [comment]
@EmanuelCampos friendly ping on the failing tests here

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 2]
```diff
@@ -793,7 +807,7 @@ def get_resource_info(self, resource_id: str, **kwargs) -> Dict:
             if (
                 self.attach_permission_metadata
             ):  # changes in access control should trigger a reingestion of the file
-                permissions = self._get_permissions_info(item)
+                permissions = self.get_permission_info(item)
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:810]: item is already a path here, but in the method above, you cast it to a path again?

Comment (logan-markewich) [reply llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:810]: Also, you call get_item_from_path above, and then call it again in get_permission_info -- did you mean to do that? (tbh, I'm surprised if this even works lol)

File: llama-index-integrations/readers/llama-index-readers-microsoft-onedrive/llama_index/readers/microsoft_onedrive/base.py [hunk 3]
```diff
@@ -653,6 +653,9 @@ def load_data(
                 f"An error occurred while loading the data: {e}", exc_info=True
             )
 
+    def get_permission_info(self, resource_id: str, *args: Any, **kwargs: Any) -> Dict:
+        return self._get_permissions_info(resource_id, *args, **kwargs)
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-microsoft-onedrive/llama_index/readers/microsoft_onedrive/base.py:657]: The types don't really make sense at all here -- resource_id is a string, item is a dict, but you treat them the same -- what is the actual type here? 😅

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py (hunk 2 above)
Comment (EmanuelCampos) [reply llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:810]: ops here is supposed to be self._get_permissions_info

**Comment by Javtor** [comment]
What's the reasoning behind adding this method? If we want to decouple permissions info from general resource info, should we still keep the logic for adding permissions info in get_resource_info?

**Comment by EmanuelCampos** [comment]
> What's the reasoning behind adding this method? If we want to decouple permissions info from general resource info, should we still keep the logic for adding permissions info in get_resource_info?

At first moment keeping it for backwards compatibility, but yes the goal is decoupling permissions from general metadata

**Review by Javtor — APPROVED** [review]

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**